File: Kth Distinct String in an Array.go
func kthDistinct(arr []string, k int) string {
    counter := make(map[string]int)
    for _, s := range arr {
        counter[s]++
    }

    for _, s := range arr {
        if counter[s] == 1 {
            k--
            if k == 0 {
                return s
            }
        }
    }
    return ""
}

File: Kth Distinct String in an Array.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        map<string, int> counter;
        for (string& s: arr) counter[s]++;
        for (string& s: arr) {
            if (counter[s] == 1) {
                if (--k == 0) return s;
            }
        }
        return "";
    }
};